### PR TITLE
feat(i18n): nav dropdown language selector

### PR DESCRIPTION
**Pull request by Spice-Z**
At https://github.com/vuejs/vitepress/pull/50, @antfu implemented i18n support, and said.

>The only thing left is nav dropdown, that would be another PR.

I created that!

Oct-01-2020 20-42-52

Working config is like this

```
...
themeConfig: {
 locales: {
    '/': {
      label: 'English',
      selectText: 'Languages',
    },
    '/zh/': {
      label: '中文',
      selectText: '选择语言',
    },
    '/ja/': {
      label: '日本語',
      selectText: '言語,
    },
  },
}
```

(To set config about site title, description, and lang, you have to use global locale config. Check https://github.com/vuejs/vitepress/pull/50)

### CODE REVIEW

File: src/client/theme-default/components/NavBarLinks.ts [hunk 1]
```diff
@@ -40,14 +41,45 @@ export default {
...
+      const candidates = localeKeys.map((v) => {
+        return {
+          text:
+            siteData.value.locales[v].label || siteData.value.locales[v].lang,
+          link: `${v}${currentContentPath}`
+        }
+      })
+      return {
+        // TODO i18n text
+        text: 'Languages',
```
Comment (Spice-Z) [inline src/client/theme-default/components/NavBarLinks.ts:69]: This text also needs i18n

Comment (Spice-Z) [reply src/client/theme-default/components/NavBarLinks.ts]: DONE

**Review by kiaking — CHANGES_REQUESTED** [review]
Thank you so much for the feature! Looking really good!

I have 2 requests.

Currently, VuePress locale config sits inside `themeConfig`. Would it be possible to move the new configs under there instead of `locales`?

```js
{
  themeConfig: {
    locales: {
      '/': {
        label: 'English',
        selectText: 'Languages',
        editLinkText: 'Edit this page on GitHub',
    },
    ...
  }
}
```

Ref: https://vuepress.vuejs.org/guide/i18n.html#default-theme-i18n-config

2. Can we move i18n dropdown before GitHub repo link at Nav?

File: src/client/theme-default/components/NavBarLinks.vue [hunk 2]
```diff
@@ -7,6 +7,7 @@
       </template>
     </template>
     <NavBarLink v-if="repoInfo" :item="repoInfo" />
+    <NavDropdownLink v-if="localeCandidates" :item="localeCandidates" />
```
Comment (kiaking) [inline src/client/theme-default/components/NavBarLinks.vue:10]: Lets move this one above the GitHub repo nav 🙏

Comment (Spice-Z) [reply src/client/theme-default/components/NavBarLinks.vue:9]: Moved!

**Comment by Spice-Z** [comment]
@kiaking 
I fixed both of your requests!

About first request, I didn't notice vitepress( and vuepress) has two kinds of locales in config...!
(`locales` is used as `Site Level i18n Config`, `themeConfig.locals` is used as `Theme i18n Config`.)

Now, this selector will use `themeConfig.locals`'s `label` and `selectText`

File: src/client/theme-default/components/NavBarLinks.ts [hunk 3]
```diff
@@ -40,14 +41,49 @@ export default {
       }
       return null
     })
+
+    const localeCandidates = computed(() => {
+      const locales = siteData.value.themeConfig.locales
+      const localeKeys = Object.keys(locales)
```
Comment (kiaking) [inline src/client/theme-default/components/NavBarLinks.ts:47]: At here, we must check whether if the `locales` exists or not.

```js
const locales = siteData.value.themeConfig.locales
if (!locales) {
  return null
}
const localeKeys = Object.keys(locales)
```

If the config doesn't contain any locales, `Object.keys(locales)` will fail due to `locales` is `undefined` in that case.

**Review by kiaking — CHANGES_REQUESTED** [review]
Thanks a lot! Just one more thing. I found a bug that this PR will not work on configs that does not contain any locales info. Please see the inline comment I made 🙏

**Comment by Spice-Z** [comment]
@kiaking 
Thank you !
Sorry to late commit..., I fixed it!

I have to think of any every case of config, or have to write test code. 😅

**Review by kiaking — APPROVED** [review]
Great! Thanks a lot. Tested at my side and it is working well 🙌

**Pull request merged**